server/src/main/java/com/distkv/server/storeserver/runtime/workerpool/Worker.java

```diff
@@ -44,28 +45,35 @@
 
 public class Worker extends Thread {
 
+  private static Logger LOGGER = LoggerFactory.getLogger(Worker.class);
+
   private StoreRuntime storeRuntime;
 
-  private static Logger LOGGER = LoggerFactory.getLogger(Worker.class);
+  private BlockingQueue<InternalRequest> queue;
+  /**
+   * A manager that handles key expire requests.
+   */
+  private ExpirationManager expirationManager;
+
+  /**
+   * Store engine.
+   */
+  private KVStore storeEngine;
 
   public Worker(StoreRuntime storeRuntime) {
     this.storeRuntime = storeRuntime;
+    storeEngine = new KVStoreImpl();
+    expirationManager = new ExpirationManager(storeRuntime.getConfig());
```

Comment:
Why create ExpirationManager in every worker? It should be in the `StoreRuntime`.
The same as `storeEngine`.

Reply:
Make sence!